### PR TITLE
ICU-20772 - Fix some minor issues

### PR DESCRIPTION
**Pull request by sylvestre**
##### Checklist

- [ ] Issue filed: [ICU-20772]
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

[ICU-20772]: https://unicode-org.atlassian.net/browse/ICU-20772

### CODE REVIEW

**Comment by sven-oly** [comment]
There are a couple of things that should be done with this PR:
a. The commit messages must include the issue information. Please see: http://site.icu-project.org/repository/gitdev#TOC-Review-commit-to-Unicode-master

b. Please take a look at the build failures and resolve them.

**Comment by sffc** [comment]
@sylvestre please take a look at Craig's comment and the build failures.

**Review by sven-oly — APPROVED** [review]
Resolving this old PR

**Comment by markusicu** [comment]
> There are a couple of things that should be done with this PR: a. The commit messages must include the issue information. Please see: http://site.icu-project.org/repository/gitdev#TOC-Review-commit-to-Unicode-master
> 
> b. Please take a look at the build failures and resolve them.

These issues are not resolved.

File: docs/userguide/icu_data/buildtool.md [hunk 1]
```diff
@@ -569,7 +569,7 @@ files are not in such a directory, derb fails with U_MISSING_RESOURCE_ERROR.
 of the filter rules matters a great deal in how effective your data size
 reduction can be, and the wildcard `*` can sometimes produce behavior that is
 tricky to reason about. For example, these three lists of filter rules look
-similar on first glance but acutally produce different output:
+similar on first glance but actually produce different output:
```
Comment (markusicu) [inline docs/userguide/icu_data/buildtool.md:572]: This was fixed a couple of years ago in PR #1506.

File: icu4c/source/common/edits.cpp [hunk 2]
```diff
@@ -482,7 +482,6 @@ UBool Edits::Iterator::next(UBool onlyChanges, UErrorCode &errorCode) {
     int32_t u = array[index++];
     if (u <= MAX_UNCHANGED) {
         // Combine adjacent unchanged ranges.
-        changed = FALSE;
```
Comment (markusicu) [inline icu4c/source/common/edits.cpp:485]: This looks bad. `changed` is a member variable, and when it's not set right, then the observable iterator state is wrong.

File: icu4c/source/common/rbbirb.cpp [hunk 3]
```diff
@@ -68,7 +68,6 @@ RBBIRuleBuilder::RBBIRuleBuilder(const UnicodeString   &rules,
     fLBCMNoChain        = FALSE;
     fLookAheadHardBreak = FALSE;
     fUSetNodes          = NULL;
-    fRuleStatusVals     = NULL;
```
Comment (markusicu) [inline icu4c/source/common/rbbirb.cpp:71]: Bad change. If something goes wrong before this field is assigned a new UVector, then the object is left with this field unassigned.

File: icu4c/source/i18n/msgfmt.cpp [hunk 4]
```diff
@@ -1639,6 +1639,7 @@ void MessageFormat::cacheExplicitFormats(UErrorCode& status) {
         Formattable::Type formattableType;
 
         switch (argType) {
+        case UMSGPAT_ARG_TYPE_SELECT:
```
Comment (markusicu) [inline icu4c/source/i18n/msgfmt.cpp:1642]: This change seems like purely a matter of taste. The switch cases are probably in the order of their enum declaration, which is also a logical grouping. Better to leave as is.

Some of these changes, if we did want to make them, should be done in parallel in both C++ and Java, to keep the code in sync.

**Review by markusicu — CHANGES_REQUESTED** [review]
The typos have been fixed by other PRs now, and the code changes are not good. Nothing left to be done here. @echeran FYI

**Pull request closed without merging**